src/object.js

```diff
@@ -38,13 +38,14 @@ export const stripEmptyObjects = _.pickBy(isNotEmptyObject)
 // { a: { b: 1, c: 2 } }, [ 'b' ] -> { a: { b: 1 } }
 export const pickInto = (map, source) => _.mapValues(pickIn(source), map)
 
-// map rename implementation (not used here yet):
-// http://jsfiddle.net/daedalus28/8uQUD/
-export const renameProperty = _.curry((from, to, target) => {
-  target[to] = target[from]
-  delete target[from]
-  return target
-})
+export const renameProperty = _.curry((from, to, target) =>
+  _.has(from, target)
+    ? _.flow(
+        x => _.set(to, _.get(from, x), x),
+        _.unset(from)
+      )(target)
+    : _.cloneDeep(target)
```

Comment:
I don't think this is necessary.

Reply:
The _.cloneDeep is necessary because otherwise you are returning a reference to the object that was passed in even if the function has not mutated the object in any way.